Source/WebKit/GPUProcess/graphics/WebGPU/RemoteQuerySet.cpp

```diff
@@ -53,7 +53,7 @@ void RemoteQuerySet::stopListeningForIPC()
 
 void RemoteQuerySet::destroy()
 {
-    m_backing->destroy();
+    m_objectHeap.removeObject(m_identifier);
```

Comment:
I think this is wrong. the destroy() function should delete _just_ the metal object, keeping the wgpu object alive

Reply:
I took your suggestion because I think it is lower risk, as with it there is no need to check the destroyed flag. 

However I think in a future patch we should tear down not just the wgpu object but also the Remote* and Remote*Proxy objects, leaving only the JS object alive. That seems more in-line with the behavior from the specification which the application loses access to the object after `destroy` is called.